client/pluginmanager/csimanager/manager.go

```diff
@@ -75,6 +75,19 @@ func (c *csiManager) PluginManager() pluginmanager.PluginManager {
 	return c
 }
 
+// WaitForPlugin waits for a specific plugin to be registered and available,
+// unless the context is canceled, or it takes longer than a minute.
+func (c *csiManager) WaitForPlugin(ctx context.Context, pType, pID string) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Minute)
+	defer cancel()
+	p, err := c.registry.WaitForPlugin(ctx, pType, pID)
+	if err != nil {
+		return fmt.Errorf("%s plugin '%s' did not become ready: %w", pType, pID, err)
+	}
+	c.ensureInstance(p)
```

Comment:
`ensureInstance` has a note that we're assuming `c.instances.Lock` is being held, so that the caller is controlling release of the lock (this allows for safe writes back to the locked resources without interleaving). So we'll need to hold the lock here too.

Reply:
Oh good lookin out.  I guess I assumed it managed the lock itself since it does the writing.